### DISCUSSION TITLE
How to pass parameters to a strategy class

### DISCUSSION THREAD

**Discussion by sjg20** [labgrid-project/labgrid#1363] [General]
All of the examples I can find have something like this in the environment file:

      ShellStrategy: {}

i.e. the properties are empty. I would like to pass some properties to a strategy from the environment file. Is that possible?

      RpiStrategy:
        board: rpi_3_32b

**Comment by Bastian-Krause** [comment]
Sure, the properties are used to instanciate the strategy. Your strategy could then look like this:

```python
@target_factory.reg_driver
@attr.s(eq=False)
class RpiStrategy(Strategy):
    bindings = {
        ...
    }

    board = attr.ib(validator=attr.validators.instance_of(str))

    ...
```

For what do you need the board type? Maybe this could be solved via place tags or feature flags?

**Comment by sjg20** [comment]
Thanks, that works well.

The board type is needed so that I know how to write U-Boot to that board (i.e. its storage device). Some use a filesystem but most use a raw device.

**Reply by jluebbe** [reply]
When the differences become larger, having two separate strategies (in the same Python file) can be less complex. Shared functionality can be in a common base class or simply as helper functions called during state transitions.

**Reply by sjg20** [reply]
OK, thanks, let's discuss that in the other issue I created
